compiler/rustc_resolve/src/macros.rs

```diff
@@ -714,15 +746,28 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 // expanded into a dummy fragment for recovery during expansion.
                 // Now, post-expansion, the resolution may succeed, but we can't change the
                 // past and need to report an error.
-                // However, non-speculative `resolve_path` can successfully return private items
+                // Special cases:
+                // 1. non-speculative `resolve_path` can successfully return private items
                 // even if speculative `resolve_path` returned nothing previously, so we skip this
                 // less informative error if the privacy error is reported elsewhere.
+                // 2. issue #118455, unresolved top level attribute error didn't imported prelude and
+                // already have emitted an error, report builtin macro and attributes error by the way,
+                // so `check_invalid_crate_level_attr` in can ignore them.
+
+                let is_builtin = res.opt_def_id().map_or(false, |_| this.is_builtin(res));
                 if this.privacy_errors.is_empty() {
-                    this.dcx().emit_err(CannotDetermineMacroResolution {
-                        span,
-                        kind: kind.descr(),
-                        path: Segment::names_to_string(path),
-                    });
+                    let mut fallback = !is_builtin;
+                    if is_builtin && krate.id == ast::CRATE_NODE_ID {
```

Comment:
`krate.id` is always `ast::CRATE_NODE_ID` (at least after expansion, and we are after expansion here).

Also, it seems like `fallback` can be simplified to a single `&&`.